### DISCUSSION TITLE
Action not updating array of object state (Vue 3)

### DISCUSSION THREAD

**Discussion by mianfg** [vuejs/pinia#1927] [Help and Questions]
I am using Vue 3 and Pinia, and I have a store `cards.js`, like this one:

```js
import { defineStore } from 'pinia'
import { useLanguageStore } from './language'
import axios from 'axios'


export const useCardsStore = defineStore({
    id: 'cards',
    state: () => ({
        cards: []
    }),
    actions: {
        async generateCards() {
            const languageStore = useLanguageStore();

            this.cards = (await axios({
                method: "post",
                url: "<endpoint>", // some endpoint
                data: {
                    text: languageStore.sentence,
                    language: languageStore.language
                }
            })).data.map(token => ({
                text: token.text,
                pos: token.pos,
            }));

            console.log(this.cards)  // see (*)
        }
    }
})
```

Then, inside a component, I use something like this:

```vue
<template>
...
   <input v-model="sentence">
   <button @click.prevent="generateCards()">Generate</button>
...
</template>

<script setup>
import { storeToRefs } from 'pinia'
import { useCardsStore } from '../stores/cards'
import { useLanguageStore } from '../stores/language'

const { cards } = storeToRefs(useCardsStore())
const { sentence } = storeToRefs(useLanguageStore())
const { generateCards } = useCardsStore()
</script>
```

However, even though when I click in the button, console output after mofifying `this.cards` (*) is:

```
Proxy {0: {…}, 1: {…}, 2: {…}, 3: {…}}
  [[Handler]]: Object
  [[Target]]: Array(4)
    0: {text: 'esto', pos: 'PRON'}
    1: {text: 'es', pos: 'VERB'}
    length: 2
  [[Prototype]]: Array(0)
  [[IsRevoked]]: false
```

(Hence the array is being correctly generated) Inside of Vue devtools I can see the state `cards` has not changed (as well as I can see it in the application).

Any help would be appreciated. Thank you.

> Note: I followed [this](https://blog.logrocket.com/complex-vue-3-state-management-pinia) tutorial.

**Comment by ghiscoding** [comment]
@mianfg have you found a way to fix this problem? I believe I have a similar issue so I'm wondering what you did? My use case is slightly different because I only update an item of the array but it also doesn't trigger a change even with `.slice()`, however doing a full assignment with `this.myArray = [...myArray]` does work but it's takes more resources to do it this way
